### PR TITLE
Big random cleanup

### PR DESCRIPTION
**Pull request by kamilritz**
This PR contains multiple non functional improvements. It can easily be reviewed commit by commit.

### CODE REVIEW

File: EKF/ekf_helper.cpp [hunk 1]
```diff
@@ -746,12 +727,14 @@ void Ekf::getAuxVelInnovRatio(float &aux_vel_innov_ratio) const
 
 void Ekf::getFlowInnov(float flow_innov[2]) const
 {
-	memcpy(flow_innov, _flow_innov, sizeof(_flow_innov));
+	flow_innov[0] = _flow_innov(0);
```
Comment (jkflying) [inline EKF/ekf_helper.cpp:730]: You can do these (and the variance, and the mag/mavVar) with `_flow_innov.copyTo(flow_innov);`

File: EKF/mag_fusion.cpp [hunk 2]
```diff
@@ -130,14 +128,14 @@ void Ekf::fuseMag()
...
 
 	// check for a badly conditioned covariance matrix
-	if (_mag_innov_var[2] >= R_MAG) {
+	if (_mag_innov_var(2) >= R_MAG) {
 		// the innovation variance contribution from the state covariances is non-negative - no fault
 		_fault_status.flags.bad_mag_z = false;
 
-	} else if (_mag_innov_var[2] > 0.0f) {
+	} else {
```
Comment (jkflying) [inline EKF/mag_fusion.cpp:138]: Are you sure this keeps the same behavior?

File: EKF/terrain_estimator.cpp [hunk 3]
```diff
@@ -222,64 +222,64 @@ void Ekf::fuseFlowForTerrain()
...
 	// calculate flow innovation (y axis)
-	_flow_innov[1] = pred_flow_y - opt_flow_rate(1);
+	_flow_innov(1) = pred_flow_y - opt_flow_rate(1);
 
 	// calculate correction term for terrain variance
 	const float KyHyP =  Ky * Hy * _terrain_var;
 
 	// innovation consistency check
-	flow_test_ratio = sq(_flow_innov[1]) / (sq(gate_size) * _flow_innov_var[1]);
+	flow_test_ratio = sq(_flow_innov(1)) / (sq(gate_size) * (1));
```
Comment (jkflying) [inline EKF/terrain_estimator.cpp:279]:
```suggestion
	flow_test_ratio = sq(_flow_innov(1)) / (sq(gate_size) * _flow_innov_var(1));
```

File: EKF/ekf_helper.cpp [hunk 4]
```diff
@@ -1139,7 +1126,7 @@ void Ekf::get_ekf_soln_status(uint16_t *status)
 	soln_status.flags.pred_pos_horiz_abs = soln_status.flags.pos_horiz_abs;
 	bool gps_vel_innov_bad = (_gps_vel_test_ratio(0) > 1.0f) || (_gps_vel_test_ratio(1) > 1.0f);
 	bool gps_pos_innov_bad = (_gps_pos_test_ratio(0) > 1.0f);
-	bool mag_innov_good = (_mag_test_ratio[0] < 1.0f) && (_mag_test_ratio[1] < 1.0f) && (_mag_test_ratio[2] < 1.0f) && (_yaw_test_ratio < 1.0f);
+	bool mag_innov_good = _mag_test_ratio.max() < 1.0f && (_yaw_test_ratio < 1.0f);
```
Comment (bresch) [inline EKF/ekf_helper.cpp:1129]:
```suggestion
	bool mag_innov_good = (_mag_test_ratio.max() < 1.0f) && (_yaw_test_ratio < 1.0f);
```

File: EKF/mag_fusion.cpp (hunk 2 above)
Comment (bresch) [reply EKF/mag_fusion.cpp:138]: It actually fixes the case where `_mag_innov_var` would suddenly go negative

File: EKF/optflow_fusion.cpp [hunk 5]
```diff
@@ -101,13 +101,11 @@ void Ekf::fuseOptFlow()
 	// calculate optical LOS rates using optical flow rates that have had the body angular rate contribution removed
 	// correct for gyro bias errors in the data used to do the motion compensation
 	// Note the sign convention used: A positive LOS rate is a RH rotation of the scene about that axis.
-	Vector2f opt_flow_rate;
-	opt_flow_rate(0) = _flow_compensated_XY_rad(0) / _flow_sample_delayed.dt + _flow_gyro_bias(0);
-	opt_flow_rate(1) = _flow_compensated_XY_rad(1) / _flow_sample_delayed.dt + _flow_gyro_bias(1);
+	const Vector2f opt_flow_rate= _flow_compensated_XY_rad / _flow_sample_delayed.dt + Vector2f(_flow_gyro_bias);
```
Comment (bresch) [inline EKF/optflow_fusion.cpp:104]:
```suggestion
	const Vector2f opt_flow_rate = _flow_compensated_XY_rad / _flow_sample_delayed.dt + Vector2f(_flow_gyro_bias);
```

File: EKF/mag_fusion.cpp (hunk 2 above)
Comment (kamilritz) [reply EKF/mag_fusion.cpp:138]: Yes, it is not completely non functional. But in a good way.

File: EKF/terrain_estimator.cpp (hunk 3 above)
Comment (kamilritz) [reply EKF/terrain_estimator.cpp:279]: Thanks, that would have been bad.

**Comment by kamilritz** [comment]
All feedback is addressed. Changes are put in the individual commits.

**Comment by kamilritz** [comment]
Not sure why the Firmware Build Test fail now when opening a PR from a fork. Will have to look at this at some point.
I quickly checked locally, and it is building with current Firmware master.

File: EKF/gps_yaw_fusion.cpp [hunk 6]
```diff
@@ -245,29 +244,20 @@ void Ekf::fuseGpsAntYaw()
...
-	// only apply covariance and state corrections if healthy
 	if (healthy) {
 		_time_last_gps_yaw_fuse = _time_last_imu;
 		// apply the covariance corrections
-		for (unsigned row = 0; row < _k_num_states; row++) {
-			for (unsigned column = 0; column < _k_num_states; column++) {
-				P(row,column) = P(row,column) - KHP(row,column);
-			}
-		}
+		P -= KHP;
```
Comment (bresch) [inline EKF/gps_yaw_fusion.cpp:259]: This is by far my favorite cleanup.

**Comment by bresch** [comment]
I'm not squashing to allow easier bisect just in case we realize that it broke something in the future.

**Pull request merged**